deployment/src/main/java/com/datastax/oss/quarkus/deployment/internal/CassandraClientProcessor.java

```diff
@@ -65,24 +65,23 @@ FeatureBuildItem feature() {
   @BuildStep
   List<ReflectiveClassBuildItem> registerGraphForReflection() {
 
-    /* TODO: We're creating an RCBI instance per class here to match up with the pre-Quarkus3.0 usage.
-     * Check to see if we can just simplify this to something like RCBI.builder(names).foo().build() */
-    return Stream.of(
-            // Required for the driver DependencyCheck mechanism
-            "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal",
-            // Should be initialized at build time:
-            "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0",
-            "org.apache.tinkerpop.shaded.jackson.databind.deser.std.StdDeserializer",
-            // Required by Tinkerpop:
-            // TODO check if this is really all that is instantiated by reflection
-            "org.apache.tinkerpop.gremlin.structure.Graph",
-            "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph",
-            "org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph",
-            "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource")
-        .map(
-            (clzName) ->
-                ReflectiveClassBuildItem.builder(clzName).methods(true).fields(true).build())
-        .collect(Collectors.toList());
+    return Collections.singletonList(
+        ReflectiveClassBuildItem.builder(
+                // Required for the driver DependencyCheck mechanism
+                "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal",
+                // Should be initialized at build time:
+                "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0",
+                "org.apache.tinkerpop.shaded.jackson.databind.deser.std.StdDeserializer",
+                // Required by Tinkerpop:
+                // TODO check if this is really all that is instantiated by reflection
+                "org.apache.tinkerpop.gremlin.structure.Graph",
+                "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph",
+                "org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph",
+                "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource")
+            .constructors(true)
```

Comment:
Looks like constructors(true) was added here when it wasn't present in the original

Reply:
Because previously constructors were true by default and that [method is now deprecated](https://github.com/quarkusio/quarkus/blob/3.0.3.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L80-L87) and hence the change.

Reply:
Before you ask, even in [2.x Quarkus version, constructors were always `true`](https://github.com/quarkusio/quarkus/blob/2.16.7.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L24-L26).

Reply:
The specific constructor you cite for 3.0 does calls another constructor in turn and it does set the "constructors" param to true when it does so.  But that's only because the constructor in question doesn't accept a "constructors" argument.  [Other](https://github.com/quarkusio/quarkus/blob/3.0.3.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L71-L78) [constructors](https://github.com/quarkusio/quarkus/blob/3.0.3.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L89-L96) do accept such an argument.

It's true that these methods are deprecated, but this appears to be the case for all RCBI constructors; based on the comments it's clear they're trying to steer developers to use the builders instead (which is entirely sensible).  The builder still very much includes a "constructors" [method](https://github.com/quarkusio/quarkus/blob/3.0.3.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L196-L199) which accepts a boolean argument but [the default](https://github.com/quarkusio/quarkus/blob/3.0.3.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L181) is still "true".

So at best including "constructors(true)" here is redundant (since it's just restating the default).

With respect to Quarkus 2.x: the set of constructors there look very similar (they just aren't deprecated yet).  As discussed above some of the constructors set "constructors" to true if they don't accept that as an argument but there are plenty of constructors in 2.x which do accept a boolean arg corresponding to "constructors"... so it very much comes down to which constructor a developer wants to make use of.

I guess I'd argue against changing the builder methods called here unless we have a clear benefit from it.  Seem sensible?

Reply:
Good point. Removed calling the `.constructors(true)` from here.